File: site/assets/experience.js
'use strict';
(() => {
  const app=document.querySelector('.football-app');
  document.querySelectorAll('.global-nav a').forEach(a=>{if(new URL(a.href).pathname===location.pathname)a.setAttribute('aria-current','page');});
  if(!app)return;
  const root=new URL('../',location.href),mode=app.dataset.mode;
  const esc=s=>String(s??'—').replace(/[&<>"']/g,c=>({'&':'&amp;','<':'&lt;','>':'&gt;','"':'&quot;',"'":'&#39;'}[c]));
  const today=new Intl.DateTimeFormat('sv-SE',{timeZone:'Asia/Tokyo'}).format(new Date());
  const count=document.getElementById('result-count');
  const status=m=>m.status==='played'?'played':m.date&&m.date<today?'awaiting':'upcoming';
  const match=m=>{const played=m.status==='played';const tie=played&&m.home_score===m.away_score&&['home','away'].includes(m.winner)?` / ${esc(m[m.winner])} タイブレーク勝利`:'';const url=m.url?new URL(m.url,root).href:m.pdf_url;return `<${url?'a':'div'} class="match-card"${url?` href="${esc(url)}"`:''}><span class="match-meta">${esc(m.date||'日程未定')} · ${esc(m.category)}</span><span class="match-teams"><span>${esc(m.home)}</span><strong>${played?`${m.home_score} – ${m.away_score}`:'VS'}</strong><span>${esc(m.away)}</span></span><span class="match-meta">${played?'試合終了':status(m)==='awaiting'?'結果反映待ち':'試合予定'}${tie}${m.venue?' / '+esc(m.venue):''}${!m.url&&url?' / 公式結果PDF':''}</span></${url?'a':'div'}>`;};
  fetch(new URL('assets/football-data.json',root)).then(r=>{if(!r.ok)throw Error();return r.json();}).then(data=>{
    if(mode==='myteam'){
      const select=document.getElementById('favorite'),out=document.getElementById('myteam-results');
      const names=[...new Set(data.teams.map(t=>t.team))].sort((a,b)=>a.localeCompare(b,'ja'));
      names.forEach(t=>select.add(new Option(t,t)));
      try{const saved=localStorage.getItem('amefootmania.myteam');if(names.includes(saved))select.value=saved;}catch{}
      function draw(save=false){const team=select.value;let stored=true;if(save)try{if(team)localStorage.setItem('amefootmania.myteam',team);else localStorage.removeItem('amefootmania.myteam');}catch{stored=false;}
        count.textContent=team?`${team}を表示しています。${stored?'':'ブラウザへの保存ができないため、この画面のみで選択を保持します。'}`:'大学を選んでください。';
        if(!team){out.innerHTML='';return;}
        const latest=Math.max(...data.matches.map(m=>m.year));const ms=data.matches.filter(m=>m.year===latest&&(m.home===team||m.away===team));
        const recent=ms.filter(m=>m.status==='played').sort((a,b)=>(b.date||'').localeCompare(a.date||'')).slice(0,3);
        const next=ms.filter(m=>status(m)==='upcoming').sort((a,b)=>(a.date||'9999').localeCompare(b.date||'9999')).slice(0,3);
        const pending=ms.filter(m=>status(m)==='awaiting');
        out.innerHTML=`<p>${data.teams.filter(t=>t.team===team).map(t=>`<a class="primary-link" href="${esc(new URL(t.url,root).href)}">${esc(t.label)}のチームページ →</a>`).join(' ')}</p><h2>次の試合</h2><div class="match-grid">${next.map(match).join('')||'<p>今後の確定日程は未掲載です。</p>'}</div><h2>直近の結果</h2><div class="match-grid">${recent.map(match).join('')||'<p>掲載結果はありません。</p>'}</div>${pending.length?`<h2>結果反映待ち</h2><div class="match-grid">${pending.map(match).join('')}</div>`:''}<p><a href="../archive/index.html?team=${encodeURIComponent(team)}&year=all">この大学の過去の記録 →</a></p>`;
      }
      select.addEventListener('change',()=>draw(true));document.getElementById('clear-favorite').addEventListener('click',()=>{select.value='';draw(true);});draw();return;
    }
    const controls=[...app.querySelectorAll('select')],params=new URLSearchParams(location.search);let limit=30;
    controls.forEach(c=>{if([...c.options].some(o=>o.value===params.get(c.id)))c.value=params.get(c.id);});
    const val=id=>document.getElementById(id)?.value||'all',results=document.getElementById('match-results'),tables=document.getElementById('standings-results'),more=document.getElementById('load-more');
    function render(){const year=mode==='archive'?val('year'):app.dataset.season,league=val('league'),team=val('team');
      let ms=data.matches.filter(m=>(year==='all'||String(m.year)===year)&&(league==='all'||m.league===league)&&(team==='all'||m.home===team||m.away===team)&&(mode==='archive'?m.status==='played':val('status')==='all'||status(m)===val('status')));
      ms.sort((a,b)=>{if(mode==='matches'&&a.status!==b.status)return a.status==='played'?1:-1;return a.status==='played'?(b.date||'').localeCompare(a.date||''):(a.date||'9999').localeCompare(b.date||'9999');});
      count.textContent=`${ms.length}試合${ms.length>limit?`（${limit}試合を表示）`:''}`;
      results.innerHTML='<h2>試合'+(mode==='archive'?'結果':'一覧')+'</h2><div class="match-grid">'+(ms.slice(0,limit).map(match).join('')||'<p class="empty-result">条件に一致する試合はありません。年度・大学・リーグを変更してください。</p>')+'</div>';
      more.hidden=ms.length<=limit;
      if(mode==='archive'){
        const blocks=data.standings.filter(b=>(year==='all'||String(b.year)===year)&&(league==='all'||b.league===league)&&(team==='all'||b.entries.some(t=>t.team===team))).sort((a,b)=>b.year-a.year);
        tables.innerHTML='<h2>順位表</h2>'+(blocks.map(b=>`<section class="archive-block"><h3>${b.year}年 · ${esc(b.block)}</h3><div class="tbl" tabindex="0" role="region" aria-label="${b.year}年 ${esc(b.block)}の順位表"><table><thead><tr><th scope="col">順位</th><th scope="col">大学</th><th scope="col">勝ち点</th><th scope="col">試合</th><th scope="col">勝–敗</th><th scope="col">得失点差（参考）</th></tr></thead><tbody>${b.entries.map(t=>`<tr${t.team===team?' class="selected-team"':''}><td>${esc(t.rank)}</td><th scope="row">${esc(t.team)}${t.team===team?'（選択中）':''}</th><td>${esc(t.points)}</td><td>${esc(t.games)}</td><td>${esc(t.wins)}–${esc(t.losses)}</td><td>${esc(t.goal_diff)}</td></tr>`).join('')}</tbody></table></div><p class="note"><a href="${esc(b.source)}">${b.year}年の公式星取表（KCFA） ↗</a></p></section>`).join('')||'<p>条件に一致する順位表はありません。</p>');
      }
    }
    function sync(){const p=new URLSearchParams();controls.forEach(c=>p.set(c.id,c.value));history.replaceState(null,'','?'+p.toString());limit=30;render();}
    controls.forEach(c=>c.addEventListener('change',sync));more.addEventListener('click',()=>{limit+=30;render();});document.getElementById('reset').addEventListener('click',()=>{controls.forEach(c=>c.selectedIndex=0);sync();});render();
  }).catch(()=>{count.textContent='データを読み込めませんでした。ページを再読み込みするか、リーグ別ページをご利用ください。';});
})();

File: pipeline/experience.py
from html import escape as e
from datetime import date
import json


def card(lg, m, prefix=''):
    played=m['status']=='played'
    score=f"{m['home_score']} – {m['away_score']}" if played else 'VS'
    status='試合終了' if played else ('結果反映待ち' if m.get('date') and m['date']<date.today().isoformat() else '試合予定')
    tie=f" / {m[m['winner']]} タイブレーク勝利" if played and m['home_score']==m['away_score'] and m.get('winner') in ('home','away') else ''
    return f'''<a class="match-card" href="{prefix}{lg['code']}/matches/{m['id']}/index.html"><span class="match-meta">{e(m.get('date') or '日程未定')} · {e(m['category'])}</span><span class="match-teams"><span>{e(m['home'])}</span><strong>{score}</strong><span>{e(m['away'])}</span></span><span class="match-meta">{status}{e(tie)}{(' / '+e(m['venue'])) if m.get('venue') else ''}</span></a>'''


def build(g, leagues, articles, meta):
    current=[(lg,m) for lg in leagues for m in lg['matches']]
    recent=sorted([(lg,m) for lg,m in current if m['status']=='played'],key=lambda x:x[1].get('date') or '',reverse=True)
    upcoming=sorted([(lg,m) for lg,m in current if m['status']=='scheduled' and m.get('date') and m['date']>=date.today().isoformat()],key=lambda x:x[1]['date'])
    team_count=len({t for lg in leagues for t in lg['teams']})
    league_cards=''.join(f'''<a class="league-card" href="{lg['code']}/index.html"><span>KANTO COLLEGE FOOTBALL</span><h3>{e(lg['label'])}</h3><p>{len(lg['teams'])}チーム · {lg['meta']['season_year']} SEASON</p><b>日程・順位・チームを見る ↗</b></a>''' for lg in leagues)
    body=f'''<div class="season-strip">COLLEGE FOOTBALL {meta['season_year']}<span>関東学生 TOP8・BIG8</span></div>
    <section class="editorial-hero"><img src="assets/hero.jpg" width="1600" height="900" alt="アメリカンフットボールのイメージ"><div class="hero-copy"><p class="eyebrow">EVERY DOWN. EVERY GAME.</p><h1>一球に、<br>すべてを。</h1><p>仲間の勝利も、ライバルの一戦も。<br>大学アメフトの熱を、ここから。</p><div class="hero-actions"><a class="primary-link" href="matches/index.html">試合結果をチェック →</a><a href="myteam/index.html">マイチームを選ぶ →</a></div><div class="hero-stats"><span><b>{len(leagues)}</b> カテゴリ</span><span><b>{team_count}</b> チーム</span><span><b>{meta['season_year']}</b> SEASON</span></div></div></section>
    <section><div class="section-heading"><div><p class="eyebrow">01 / MATCH CENTER</p><h2>試合の熱を、追いかけよう。</h2></div><a href="matches/index.html">全日程・結果 →</a></div><h3>最新の掲載結果</h3><div class="match-grid">{''.join(card(lg,m) for lg,m in recent[:4]) or '<p>掲載結果はまだありません。</p>'}</div><h3>次のキックオフ</h3><div class="match-grid">{''.join(card(lg,m) for lg,m in upcoming[:2]) or '<p>今後の確定日程は未掲載です。</p>'}</div><p class="note">情報更新：{e(meta['fetched_at'][:10])}。連盟公表データを掲載しています。ライブ速報ではありません。</p></section>
    <section id="leagues"><div class="section-heading"><div><p class="eyebrow">02 / THE LEAGUES</p><h2>あなたのリーグは、ここに。</h2></div></div><div class="league-grid">{league_cards}</div><p class="note">掲載対象は関東学生TOP8・BIG8です。ブロック名・年度は各表で確認できます。</p></section>
    <section class="database-promo"><p class="eyebrow">03 / FOOTBALL DATABASE</p><h2>あの対戦を、もう一度。</h2><p>年度・リーグ・大学から試合結果と順位表を探す。</p><a class="primary-link" href="archive/index.html">過去の記録を調べる →</a></section>
    <section><div class="section-heading"><div><p class="eyebrow">04 / JOURNAL</p><h2>アメフトを、もっと深く。</h2></div><a href="articles/index.html">読みもの一覧 →</a></div><div class="digest">{''.join(g.article_card(a,'') for a in articles[:3])}</div><p><a href="articles/american-football-watching-guide/index.html">はじめての観戦ガイド →</a></p></section>'''
    g.write_page('',g.page('',f'{g.SITE_NAME} | 大学アメフトの試合結果・順位・データ',body,meta))
    options=''.join(f'<option value="{lg["code"]}">{e(lg["label"])}</option>' for lg in leagues)
    teams=sorted({t for lg in leagues for _,ms in lg['matches_by_year'] for m in ms for t in (m['home'],m['away'])})
    team_options=''.join(f'<option>{e(t)}</option>' for t in teams)
    years=sorted({y for lg in leagues for y,_ in lg['matches_by_year']},reverse=True)
    dataset={'matches':[],'standings':[],'teams':[]}
    for lg in leagues:
        for year,ms in lg['matches_by_year']:
            for m in ms:
                dataset['matches'].append(dict(m,year=year,league=lg['code'],league_label=lg['label'],url=f"{lg['code']}/matches/{m['id']}/index.html" if year==lg['meta']['season_year'] else None))
        for year,standings in [(lg['meta']['season_year'],lg['standings'])]+[(h['year'],h['standings']) for h in lg['hist']]:
            for block,entries in standings.items():
                dataset['standings'].append({'year':year,'league':lg['code'],'block':block,'entries':entries,'source':f'https://www.kcfa.jp/result_team/?season={year}'})
        for t,info in lg['teams'].items():
            dataset['teams'].append({'team':t,'league':lg['code'],'label':lg['label'],'url':f"{lg['code']}/clubs/{info['slug']}/index.html"})
    (g.SITE/'assets'/'football-data.json').write_text(json.dumps(dataset,ensure_ascii=False),encoding='utf-8')
    for archive in [False,True]:
        slug='archive' if archive else 'matches'
        title='データベース' if archive else '試合・結果'
        year_options=''.join(f'<option value="{y}">{y}年</option>' for y in years)
        filters=(f'<label>年度<select id="year">{year_options}<option value="all">すべての年度</option></select></label>' if archive else '')
        filters+=f'<label>リーグ<select id="league"><option value="all">すべてのリーグ</option>{options}</select></label><label>大学<select id="team"><option value="all">すべての大学</option>{team_options}</select></label>'
        if not archive: filters+='<label>表示<select id="status"><option value="all">すべて</option><option value="upcoming">今後の試合</option><option value="played">試合結果</option><option value="awaiting">結果反映待ち</option></select></label>'
        fallback=''.join(f'<li><a href="../{lg["code"]}/{"records" if archive else "schedule"}/index.html">{lg["label"]}の{"記録室" if archive else "日程・結果"}</a></li>' for lg in leagues)
        body=f'''<p class="eyebrow">{'FOOTBALL DATABASE' if archive else 'MATCH CENTER'}</p><h1>{title}</h1><p>{'過去の試合結果と順位表を、このページで。' if archive else '関東学生TOP8・BIG8の試合を探す。'}</p><div class="football-app" data-mode="{slug}" data-season="{meta['season_year']}"><div class="filters">{filters}<button id="reset" type="button">条件をリセット</button></div><p id="result-count" role="status">データを読み込んでいます…</p><div id="standings-results"></div><div id="match-results"></div><button id="load-more" type="button" hidden>さらに30試合を表示</button></div><p class="note">更新：{e(meta['fetched_at'][:10])}。順位・勝ち点は連盟公表値、得失点差は試合結果からの参考集計です。同点スコアの決着は勝敗情報で表示します。</p><details><summary>リーグ別ページで見る</summary><ul>{fallback}</ul></details>'''
        g.write_page(slug,g.page('../',title+' | '+g.SITE_NAME,body,meta,path=slug+'/'))
    body='''<p class="eyebrow">MY TEAM</p><h1>いつも追いかける、あのチーム。</h1><p>大学を選ぶと、直近の結果と次の試合がすぐに見られます。選択はこのブラウザに保存されます。</p><div class="football-app" data-mode="myteam"><label>応援する大学<select id="favorite"><option value="">大学を選択</option></select></label><button id="clear-favorite" type="button">選択を解除</button><p id="result-count" role="status"></p><div id="myteam-results"></div></div>'''
    g.write_page('myteam',g.page('../','マイチーム | '+g.SITE_NAME,body,meta,path='myteam/'))
